finetune/base.py

```diff
@@ -62,60 +48,34 @@ def __init__(self, config=None, **kwargs):
         if self.config.num_layers_trained != self.config.n_layer and self.config.train_embeddings:
             raise ValueError("If you are only finetuning a subset of the layers, you cannot finetune embeddings.")
 
-        self.label_encoder = None
+        self.input_pipeline = self._get_input_pipeline()
         self._initialize()
-        self.target_dim = None
-        self._load_from_file = False
-        self.pad_idx_ = None
+
+    @abstractmethod
+    def _get_input_pipeline(self):
+        pass
 
     def _initialize(self):
         # Initializes the non-serialized bits of the class.
         self._set_random_seed(self.config.seed)
-
+        self.estimator_ = None
         download_data_if_required()
-        self.encoder = TextEncoder()
-
-        # symbolic ops
-        self.logits = None  # classification logits
-        self.target_loss = None  # cross-entropy loss
-        self.lm_losses = None  # language modeling losses
-        self.lm_predict_op = None
-        self.train = None  # gradient + parameter update
-        self.features = None  # hidden representation fed to classifier
-        self.summaries = None  # Tensorboard summaries
-        self.gradient_summaries = None
-        self.train_writer = None
-        self.valid_writer = None
-        self.predict_params = None
-        self.train_op = None
-        self.predict_op = None
-        self.predict_proba_op = None
-        self.sess = None
-        self.noop = tf.no_op()
-
-        # loss variables
-        self.ema_train_loss = None
-        self.ema_validation_loss = None
-        self.validation_losses = [] # running window of validation losses
-        self.best_validation_loss = float('inf') # lowest observed validation loss
-        
-        # indicator vars
-        self.is_built = False    # has tf graph been constructed?
-        self.is_trained = False  # has model been fine-tuned?
-        self.require_lm = False
+        self.estimator_dir = tempfile.mkdtemp()
```

Comment:
Might be nice to have this be something set in config that defaults to a tempdir if not explicitly set.

Reply:
Remove only ckpt files.